testing/cpm/verify_generated_pins/CMakeLists.txt

```
@@ -62,6 +62,12 @@ foreach(proj IN LISTS projects-to-verify)
     )
   endif()
 
+  if(proj IN_LIST projects-with-subdirs)
+    if(NOT pin_${proj}_src_dir)
+      message(FATAL_ERROR "pinned ${proj} should have a `source_subdir` entry")
+    endif()
+  endif()
```

This can be condensed:

```suggestion
  if(proj IN_LIST projects-with-subdirs AND NOT pin_${proj}_src_dir)
    message(FATAL_ERROR "pinned ${proj} should have a `source_subdir` entry")
  endif()
```